### ISSUE TITLE
Why listen to `change` event on collection?

### ISSUE THREAD

**Issue by mir3z** [magalhas/backbone-react-component#96]
What's the reason behind listening to `change` event on collections? This causes a lot of events which leads to re-rendering components. Wouldn't be better to listen to `add` and `remove` events?

**Comment by mistakevin** [comment]
Or even the `update` event, which only fires once for any group `add`/`remove` events

**Comment by magalhas** [comment]
@mir3z  I've been off Backbone development for quite a while, but I think the `change` event is triggered when a model attribute change, it's not related to the addition and removal of models from a collection. Or am I wrong?

Actually we're already listening to the `update` event (which listens to `add` and `remove` event) [here](https://github.com/magalhas/backbone-react-component/blob/master/lib/component.js#L333)

**Comment by mir3z** [comment]
I understand your point but let's consider the following use case.

Let's have a collection of of TODOs. Then we create a React component which represents a list of TODOs. This component uses a collection as a data source. The list consists of specific TODO items. Each item is another React component and has TODO model as a data source.

Now let's assume a model changes. It triggers `change` event. React item component re-renders to reflect model state. Then `change` event is propagates to collection. But list component listens to `change` event of collection as well so React tries to re-render whole list (at least at the virtual dom level).

My point is that `backbone-react-component` should inform React only about adding/removing items in collection since in most cases collection is rendered as a some kind of list component. And any change in model should not trigger re-rendering of whole list as it is now. Now you can achieve this only by implementing `shouldComponentUpdate`.

Your approach is fine as long you work with small collection collections and models. But it does not scale well if collection has 1000 or more heavy models.

**Comment by magalhas** [comment]
I totally agree with you and that's also the best way of binding your data sources. Feel free to PR this as well, I'll release a new minor afterwards. I think tests will need some changes to accommodate this.

#79 might take some time or may never be coded, at least by me :)

**Comment by mir3z** [comment]
Unfortunately I don't have enough time to implement this change. It seems to be rather a massive change.

**Comment by magalhas** [comment]
I don't think this is that massive change but I'll try to tackle this when I find sometime. Though I'm not sure if I'll have anytime soon
